Sources/SwiftDocC/Infrastructure/Link Resolution/DocumentationCacheBasedLinkResolver.swift

```diff
@@ -0,0 +1,540 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SymbolKit
+
+// This code is extracted from the DocumentationContext to make it easier to isolate and replace.
+
+/// A type that encapsulates resolving links by looking up references in a documentation cache.
+final class DocumentationCacheBasedLinkResolver {
```

Comment:
for conciseness, would renaming `LocalReferenceResolver` be accurate? That way we have "external resolvers" and "local resolvers".

Reply:
The `DocumentationCacheBasedLinkResolver` is currently responsible for calling the external resolver when it encounters an external reference, so in the current implementation it is responsible for resolving all references, local and external. 

I would shorten the name if it I didn't plan on following this up with another PR that adds another `...BasedLinkResolver`.